utils/utils.js

```diff
@@ -3,6 +3,70 @@ const Discord = require("discord.js"),
 
 const errorLog = new Discord.WebhookClient(config.error);
 
+/**
+ * @typedef {Object} ParsedInteraction
+ * @property {String} command - The command issued, represented as a string. 
+ * @property {Array} data - Associated data for the command, such as command options or values selected.
+ */
+
+/**
+ * Converts an interaction into a more universal format for error messages.
+ * @param {Discord.Interaction} inter The interaction to be parsed.
+ * @returns {ParsedInteraction} The interaction after it has been broken down.
+ */
+function parseInteraction(inter) {
+  if (inter.isCommand()) {
+    let commandParts = [`/${inter.commandName}`];
+    let options = inter.options;
+    if (options.data.length == 0) {
+      return {
+        command: commandParts.join(" "),
+        data: options.data
+      };
+    }
+    
+    if (options.data[0].type == "SUB_COMMAND_GROUP") {
+      commandParts.push(options.data[0].name);
+      options = options.data[0].options
+      if (options.data.length == 0) {
+        return {
+          command: commandParts.join(" "),
+          data: options.data
+        };
+      }
+    }
+
+    if (options.data[0].type == "SUB_COMMAND") {
+      commandParts.push(options.data[0].name);
+      options = options.data[0].options
+      return {
+        command: commandParts.join(" "),
+        data: options.data
+      };
+    }
+  }
```

Comment:
I believe `options.data` nests if it's a sub command group or sub command. For example:
```js
options.data = [{
  type: "SUB_COMMAND_GROUP",
  name: "game",
  options: [{
    type: "SUB_COMMAND",
    name: "redeem",
    options: [{
      name: "code",
      type: "STRING",
      value: "SOMECODE"
    }]
  }]
}]
```
If it's nested like I think it is, it'll mess up parsing down around lines 173

Reply:
See lines 30 and 41 that are intended to address this. Do you think those lines will do the job?

Reply:
I mean ... if you want to get *technical* about it! I overlooked that.